### PR TITLE
New application: io.github.TheWisker.Cavasik

### PR DESCRIPTION
**Pull request by TheWisker**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

**Comment by TheWisker** [comment]
I think I have done everything correctly. It is my first time uploading an app to Flathub, so if anything is wrong in my procedures notify me.

File: io.github.TheWisker.Cavasik.json [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	    ]
+	},
+        {
+            "name" : "cavasik",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "dir",
+                    "path" : "."
```
Comment (JakobDev) [inline io.github.TheWisker.Cavasik.json:78]: That will not work. You have to download your source from the Internet like the other modules.

File: io.github.TheWisker.Cavasik.json [hunk 2]
```diff
@@ -0,0 +1,83 @@
+{
+    "app-id" : "io.github.TheWisker.Cavasik.desktop",
```
Comment (hfiguiere) [inline io.github.TheWisker.Cavasik.json:2]: there is never `.desktop` in the app-id.

```suggestion
    "app-id" : "io.github.TheWisker.Cavasik",
```

File: io.github.TheWisker.Cavasik.json [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        "/man",
+        "/share/doc",
+        "/share/gtk-doc",
+        "/share/man",
+        "/share/pkgconfig",
+        "*.la",
+        "*.a"
+    ],
+    "modules" : [
+    	"shared-modules/linux-audio/fftw3f.json",
```
Comment (hfiguiere) [inline io.github.TheWisker.Cavasik.json:26]: you haven't added the shared-modules.

Comment (TheWisker) [reply io.github.TheWisker.Cavasik.json:26]: What do you mean exactly?, I only use that one module.

File: io.github.TheWisker.Cavasik.json (hunk 2 above)
Comment (TheWisker) [reply io.github.TheWisker.Cavasik.json:2]: Thanks. Fixed it!

File: io.github.TheWisker.Cavasik.json (hunk 3 above)
Comment (hfiguiere) [reply io.github.TheWisker.Cavasik.json:26]: From the README on the repo:

https://github.com/flathub/shared-modules/

image

Comment (hfiguiere) [reply io.github.TheWisker.Cavasik.json:26]: you didn't do that.

Comment (JakobDev) [reply io.github.TheWisker.Cavasik.json:26]: You have to add https://github.com/flathub/shared-modules as submodule, or the builder will not find the linked modules

Comment (TheWisker) [reply io.github.TheWisker.Cavasik.json:26]: Ok, on it.

**Comment by TheWisker** [comment]
Is everything correct as of now?

**Comment by JakobDev** [comment]
It looks correct. Let's see, if the Build works.

**Comment by TheWisker** [comment]
As it seems to be the problem I will change the SVG size, will contact when it's done. Thanks 👍

**Comment by TheWisker** [comment]
The problem with the SVG should be fixed as of now. Thanks for everything.

**Comment by JakobDev** [comment]
You can trigger the test build by yourself, if you want

**Comment by JakobDev** [comment]
bot, build io.github.TheWisker.Cavasik

**Comment by JakobDev** [comment]
You should update the Commit in your Manifest

**Comment by TheWisker** [comment]
Just updated it. Now it should be everything correct. At least I hope so. Thanks for the help and sorry for my dumb mistakes. 👍

**Comment by JakobDev** [comment]
bot, build io.github.TheWisker.Cavasik

**Comment by JakobDev** [comment]
It's `GPL-3.0`, not `GNUv3`

**Comment by JakobDev** [comment]
As I said, you can trigger the build by yourself

**Comment by JakobDev** [comment]
bot, build io.github.TheWisker.Cavasik

**Comment by JakobDev** [comment]
Wait for barthalion

**Comment by TheWisker** [comment]
barthalion?

**Comment by hfiguiere** [comment]
what should have been said is that you need some patience. Pinging people doesn't make it go faster.

**Comment by barthalion** [comment]
bot, build io.github.TheWisker.Cavasik

**Comment by JakobDev** [comment]
Are you sure this is a Fork? The App you linked is written in C#, while this App is written in Python.

**Comment by TheWisker** [comment]
Yes. I was going to do a small contribution at first because I wanted to change a specific thing. But just when I was about to make the contribution they archived the [original repository](https://github.com/fsobolev/cavalier) so it would not let me. Then I saw they moved to [another repository ](https://github.com/NickvisionApps/Cavalier) which was written in another language C# (instead of Python) relative to the changes I had made. So unable to make the contributions I decided to full fork it and then proceeded to make further changes to the application. Hope this satisfies your curiosity.

**Comment by barthalion** [comment]
That's fair, I haven't noticed the Nick's fork is actually it's a full rewrite.

/merge

**Pull request closed without merging**